.github/workflows/quick.yaml

```diff
@@ -94,29 +101,43 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04 ]
+        os:
+          - ubuntu-22.04
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+        layer:
+          - { name: layer-00-default, nick: default }
+          - { name: layer-01-minimal, nick: minimal }
+          - { name: layer-02-maximus, nick: maximus }
 
```

Comment:
> PR description: `Backport of PR #1917`

Please use commit SHAs rather than GitHub PR numbers in such references.

PR changes contain more than one backport. I have not checked carefully, but I suspect most of the following master/v7 commits were backported in this v6 PR:

* 240efcbb: Use GitHub Actions for more OS-specific build tests
* 82e18891: Bootstrap libltdl to fix libtool v2.4 + automake v1.17 build
* 7ccb183a: CI: Upgrade GitHub upload-artifact action to v4
* e68abfce: CI: Use packages from GitHub Actions MacOS runner base-image
* 8805c474: CI: Upgrade GitHub Setup Node and CodeQL actions to Node 20
* 5a7811af: CI: Add clang to GitHub Actions build-tests matrix
* 45eabfa1: CI: Build layers independently to speed up GitHub Actions tests
* a9eda969: CI: better parallelism for functionality-test build
* c0cda58f: Add MacOS to Github Actions CI tests
* 2ca41d3e: CI: Update GitHub actions/checkout to v4

Reply:
Maybe the word "backport" is misleading and I should call it "port" instead? Point being, this brings to the
v6 branch something that we know to be working in v7, skipping all the intermediate steps.

Reply:
> Maybe the word "backport" is misleading 

It is not that word. It is the phrase "backport of X". This PR contains a lot more fairly important changes than a (back)port of X.

> and I should call it "port" instead?

There would be no difference as far as this change request is concerned.

> this brings to the v6 branch something that we know to be working in v7, skipping all the intermediate steps.

"_something_ that works in v7" is too vague, but I bet that you meant something closer to "_everything_ that works in v7" (within GitHub Actions configuration context, of course). FWIW, to reflect what is actually happening in this PR, I would do something like this:

```
CI: Sync GitHub Actions config with master as of commit 240efcb

----

Backported all master/v7 GitHub Actions changes as of master/v7 commit
240efcb, except those related to master-specific CodeQL tests.
```

Reply:
I see what you mean now: https://github.com/kinkie/squid/actions/runs/11619146161 shows MacOS support is missing some bits.

Reply:
Other parts (e.g Bootstrap libltdl to fix libtool v2.4 + automake v1.17 build) have already been backported to v6

Reply:
> Other parts (e.g Bootstrap libltdl to fix libtool v2.4 + automake v1.17 build) have already been backported to v6

The wording I [suggested](https://github.com/squid-cache/squid/pull/1922#discussion_r1824899714) is compatible with that fact.

That wording would have to be adjusted a bit if you decide not to add MacOS tests per your other [comment](https://github.com/squid-cache/squid/pull/1922#discussion_r1825166212):

```
CI: Sync GitHub Actions config with master as of commit 240efcb

----

Backported all master/v7 GitHub Actions changes as of master/v7 commit
240efcb, except those related to master-specific CodeQL and MacOS tests.
```